test/test_dlpack.py

```diff
@@ -144,6 +144,17 @@ def __dlpack__(self, stream=None):
             x = DLPackTensor(make_tensor((5,), dtype=torch.float32, device=device))
             from_dlpack(x)
 
+    @skipMeta
+    @onlyCUDA
+    @skipCUDAIfRocm
+    def test_dlpack_convert_default_stream(self, device):
+        x = torch.zeros(2**30, device=device)
+        torch.cuda.default_stream().synchronize()
+        self.assertTrue(torch.cuda.default_stream().query())
+        d = x.__dlpack__(1)
+        # check that the default stream has work (a pending cudaStreamWaitEvent)
+        self.assertFalse(torch.cuda.default_stream().query())
```

Comment:
I think, based on the logic `x` is on the default stream and `d` is on the default stream, so event creation and `cudaStreamWaitEvent` isn't even supposed to happen, right? I think, what ends up happening there is not actually `cudaStreamWaitEvent`, but something related to this line: https://github.com/pytorch/pytorch/pull/101318/files#diff-41eb33210e32f7a565a420a78573740a643b5f124726e7013e8b96aad158948eR1351 . Feel free to correct me

Reply:
Maybe it would make sense to create `x` on a separate stream, and call `d = x.__dlpack__(1)` from the default stream, then `cudaStreamWaitEvent` should happen on the default stream

Reply:
`x` wouldn't be created on the default stream as a quirk of test semantics---the creation should happen on a side stream IIRC.

Reply:
interesting, I didn't know that. Thanks for clarification. Is there any point in docs I can read about this behavior?

Reply:
I'm not sure where this is documented, as I learned it probably from either this comment or a similar one: https://github.com/pytorch/pytorch/blob/df9c897dc38583e6ee55a5078893e521be2e56d5/test/test_dlpack.py#L142

This property can be verified by just printing `torch.cuda.current_stream()` in a test.

Reply:
Given how confusing this is, can you please leave a comment in the test itself saying that the test runs on non-default stream?